### PR TITLE
feat[rust, python]: add `float_precision` parameter to `DataFrame.write_csv`

### LINKED ISSUE #4279: write_csv needs precision and/or format input for floating point numbers
Both `numpy` and `pandas` have input for format in their writing data methods (`fmt='%.18e'` in `numpy.savetxt` and `float_format=None` in `pandas.DataFrame.to_csv`). It would be nice to specify the format and precision of floating point numbers when saving files.

### PR DESCRIPTION
**Pull request by matteosantama**
Closes #4279 and #1125

Unfortunately `lexical` does not provide a way to specify the number of decimals to write (see [this](https://github.com/Alexhuszagh/rust-lexical/issues/34#issuecomment-838856451) comment) but we still use the fast-path when no `float_precision` is specified (by default).

### CODE REVIEW

File: polars/polars-io/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![feature(test)]
```
Comment (ritchie46) [inline polars/polars-io/src/lib.rs:1]: I don't think we should add anothe unstable feature.

File: polars/polars-io/src/csv/mod.rs [hunk 2]
```diff
@@ -74,3 +74,34 @@ use crate::mmap::MmapBytesReader;
 use crate::predicates::PhysicalIoExpr;
 use crate::utils::resolve_homedir;
 use crate::{RowCount, SerReader, SerWriter};
```
Comment (ritchie46) [inline polars/polars-io/src/csv/mod.rs:76]: Can we remove this bench. I don't thywe should have it in the code base.

File: polars/polars-io/Cargo.toml [hunk 3]
```diff
@@ -45,7 +45,7 @@ arrow = { package = "arrow2", git = "https://github.com/jorgecarleitao/arrow2",
 csv-core = { version = "0.1.10", optional = true }
 dirs = "4.0"
 flate2 = { version = "1", optional = true, default-features = false }
-lexical = { version = "6", optional = true, default-features = false, features = ["std", "parse-floats", "parse-integers"] }
+lexical = { version = "6", optional = true, default-features = false, features = ["std", "parse-floats", "parse-integers", "write-floats"] }
```
Comment (ritchie46) [inline polars/polars-io/Cargo.toml:48]: Do we need the extra feature?

**Review by ritchie46 — COMMENTED** [review]
Thanks. I have a few minor remarks.

**Comment by matteosantama** [comment]
Removed all benchmarking code

**Pull request merged**